examples/plot_scalable_poly_kernels.py

```diff
@@ -0,0 +1,191 @@
+    results[f"LSVM + PS({n_components})"] = \
+        {"time": ps_lsvm_time, "score": ps_lsvm_score}
+    print(f"Linear SVM score on {n_components} PolynomialSampler " +
+          f"features: {ps_lsvm_score:.2f}%")
+
+##############################################################################
+# Train a kernelized SVM to see how well :class:`PolynomialSampler`
+# is approximating the performance of the kernel. This, of course, may take
+# some time, as the SVC class has a relatively poor scalability. This is the
+# reason why kernel approximators are so useful:
+
+from sklearn.svm import SVC
+
+ksvm = SVC(C=500., kernel="poly", degree=4, coef0=0, gamma=1.)
+
+start = time.time()
+ksvm.fit(X_train, y_train)
+ksvm_time = time.time() - start
+ksvm_score = 100 * ksvm.score(X_test, y_test)
+
+results["KSVM"] = {"time": ksvm_time, "score": ksvm_score}
+print(f"Kernel-SVM score on raw featrues: {ksvm_score:.2f}%")
+
+##############################################################################
+# Finally, plot the resuts of the different methods against their training
+# times. As we can see, the kernelized SVM achieves a higher accuracy,
+# but its training time is much larger and, most importantly, will grow
+# much faster if the number of training samples increases.
+
+plt.figure(figsize=(7, 7))
+plt.scatter([results["LSVM"]["time"], ], [results["LSVM"]["score"], ],
+            label="Linear SVM", c="green", marker="^")
+
+plt.scatter([results["LSVM + PS(250)"]["time"], ],
+            [results["LSVM + PS(250)"]["score"], ],
+            label="Linear SVM + PolynomialSampler", c="blue")
+plt.scatter([results["LSVM + PS(500)"]["time"], ],
+            [results["LSVM + PS(500)"]["score"], ],
+            c="blue")
+plt.scatter([results["LSVM + PS(1000)"]["time"], ],
+            [results["LSVM + PS(1000)"]["score"], ],
+            c="blue")
+plt.scatter([results["LSVM + PS(2000)"]["time"], ],
+            [results["LSVM + PS(2000)"]["score"], ],
+            c="blue")
+plt.annotate("n_comp.=250", (results["LSVM + PS(250)"]["time"],
+                             results["LSVM + PS(250)"]["score"]),
+             xytext=(-30, 10), textcoords="offset pixels")
+plt.annotate("n_comp.=500", (results["LSVM + PS(500)"]["time"],
+                             results["LSVM + PS(500)"]["score"]),
+             xytext=(-30, 10), textcoords="offset pixels")
+plt.annotate("n_comp.=1000", (results["LSVM + PS(1000)"]["time"],
+                              results["LSVM + PS(1000)"]["score"]),
+             xytext=(-30, 10), textcoords="offset pixels")
+plt.annotate("n_comp.=2000", (results["LSVM + PS(2000)"]["time"],
+                              results["LSVM + PS(2000)"]["score"]),
+             xytext=(-30, 10), textcoords="offset pixels")
+
+plt.scatter([results["KSVM"]["time"], ], [results["KSVM"]["score"], ],
+            label="Kernel SVM", c="red", marker="x")
+
+plt.xlabel("Training time (s)")
+plt.ylabel("Accurary (%)")
+plt.legend()
+plt.show()
```

Comment:
The matplotlib object API is preffered as it makes it easier to manipulate individual figures https://matplotlib.org/3.3.0/api/index.html#usage-patterns

Also some parametrization,

```suggestion
fig, ax = plt.subplots(figsize=(7, 7))
ax.scatter([results["LSVM"]["time"], ], [results["LSVM"]["score"], ],
            label="Linear SVM", c="green", marker="^")

ax.scatter([results["LSVM + PS(250)"]["time"], ],
            [results["LSVM + PS(250)"]["score"], ],
            label="Linear SVM + PolynomialSampler", c="blue")
for n_components in [250, 500, 1000, 2000]:  # preferably define N_COMPONENTS above
    ax.scatter([results[f"LSVM + PS({n_components})"]["time"], ],
               [results[f"LSVM + PS({n_components})"]["score"], ],
               c="blue")
	ax.annotate(f"n_comp.={n_components}",
				 (results[f"LSVM + PS({n_components})"]["time"],
                  results[f"LSVM + PS({n_components})"]["score"]),
            	 xytext=(-30, 10), textcoords="offset pixels")

ax.scatter([results["KSVM"]["time"], ], [results["KSVM"]["score"], ],
            label="Kernel SVM", c="red", marker="x")

ax.set_xlabel("Training time (s)")
ax.set_ylabel("Accurary (%)")
ax.legend()
plt.show()
```